### PR TITLE
feat: Add RandomEffectController

### PR DESCRIPTION
**Pull request by st-pasha**
# Description

Add `RandomEffectController` -- a controller that makes the duration of some other effect controller random. This could be useful if the user wants to inject some amount of unpredictability into the effect.

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors and are passing (See [Contributor Guide]).
- [x] My PR does not decrease the code coverage, or I have __a very special case__ and explained on the PR description why this PR decreases the coverage.
- [x] I updated/added relevant documentation (doc comments with `///`) and updated/added examples in `doc/examples`.
- [x] I have formatted my code with `flutter format` and the `flutter analyze` does not report any problems.
- [x] I read and followed the [Flame Style Guide].
- [x] I have added a description of the change under `[next]` in `CHANGELOG.md`.
- [x] I removed the `Draft` status, by clicking on the `Ready for review` button in this PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

- [x] No, this is *not* a breaking change.

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md

### CODE REVIEW

File: doc/effects.md [hunk 1]
```diff
@@ -42,6 +41,7 @@ There are multiple effects provided by Flame, and you can also
 - [`SizeEffect.by`](#sizeeffectby)
 - [`SizeEffect.to`](#sizeeffectto)
 - [`OpacityEffect`](#opacityeffect)
+- [`ColorEffect`](#coloreffect)
```
Comment (spydon) [inline doc/effects.md:44]: They were ordered alphabetically before, isn't that better, or what is the reason for this change?

**Review by spydon — APPROVED** [review]

File: doc/effects.md (hunk 1 above)
Comment (st-pasha) [reply doc/effects.md:44]: Huh? They were never alphabetic: for example, Opacity < Size.
They are roughly ordered by category: first, transform effects, then visual effects, then miscellaneous effects. Maybe we should have a two-level list here to make the structure more obvious.

Comment (spydon) [reply doc/effects.md:44]: Ah alright, they were pretty close to alphabetical so that's why I assumed they were. Your ordering is fine.

**Review by spydon — APPROVED** [review]

**Review by erickzanardo — APPROVED** [review]

File: packages/flame/lib/src/effects/controllers/random_effect_controller.dart [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+      'Random generator produced a negative value: $duration',
+    );
+    child.duration = duration;
+  }
+}
+
+/// [RandomVariable] is an object capable of producing random values with the
+/// prescribed distribution function. Each distribution is implemented within
+/// its own derived class.
+abstract class RandomVariable {
```
Comment (luanpotter) [inline packages/flame/lib/src/effects/controllers/random_effect_controller.dart:100]: I have a feeling this could be useful in a general context, not only to random effect controller;
For example, on Gravity I introduced a [method similar to Uniform Random Variable](https://github.com/bluefireteam/gravitational_waves/blob/master/game/lib/game/collections.dart#L4)
If so, we could move it to its own file and add factory methods to create both private implementations.
Thoughts?

**Review by luanpotter — APPROVED** [review]
LGTM

File: packages/flame/lib/src/effects/controllers/random_effect_controller.dart (hunk 2 above)
Comment (spydon) [reply packages/flame/lib/src/effects/controllers/random_effect_controller.dart:100]: Probably as a follow-up though?
